src/test/java/guitests/PersonEditGuiTest.java

```diff
@@ -42,35 +46,143 @@ public void editPerson() {
         editPersonDialog.enterNewValues(newAlice);
         editPersonDialog.pressEnter();
 
-        assertTrue(descOnFail(alicePersonCard, newAlice), alicePersonCard.isSamePerson(newAlice));
-        //TODO: write a more elegant function that uses assertEquals but compare two different types of objects
+        mainGui.sleep(1, TimeUnit.SECONDS);
+
+        //Confirm pending state correctness
+        assertTrue(alicePersonCard.isPendingStateRootVisible());
+        assertTrue(alicePersonCard.isPendingStateLabelVisible());
+        assertFalse(alicePersonCard.isPendingStateProgressIndicatorVisible());
+        assertTrue(alicePersonCard.getPendingStateLabel().equals("Editing"));
+
+        //Confirm the right card is selected after the edit
+        assertEquals(alicePersonCard, newAlice);
+        assertEquals(1, personListPanel.getSelectedCards().size());
+        assertEquals(alicePersonCard, personListPanel.getSelectedCards().get(0));
```

Comment:
These lines too can be abstracted into a method in *GuiHandle class.
e.g. `assertTrue(personListPanel.isSelectedOnly(alice))`